File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    tangerine.sort()
    size_lst = []
    idx = 1
    cnt = 1
    exValue = tangerine[0]
    while idx < len(tangerine):
        if tangerine[idx] != exValue:
            exValue = tangerine[idx]
            size_lst.append(cnt)
            cnt = 1
        else:
            cnt += 1
        idx += 1
    size_lst.append(cnt)
    size_lst.sort(reverse=True)
    idx = 0
    while k > 0 and idx < len(size_lst):
        k -= size_lst[idx]
        idx+=1
        answer+=1

    return answer

File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(int k, vector<int> tangerine) {
    int mx = *max_element(tangerine.begin(), tangerine.end());
    vector<int> v(mx+1, 0);
    for (auto t:tangerine) {
        v[t]++;
    }
    sort(v.rbegin(), v.rend());
    int idx = 0;
    while (k > 0) {
        k -= v[idx];
        idx++;
    }
    
    return idx;
}